? libao2/ao_darwin.c
Index: configure
===================================================================
RCS file: /cvsroot/mplayer/main/configure,v
retrieving revision 1.887
diff -u -r1.887 configure
--- configure	17 Jul 2004 09:53:35 -0000	1.887
+++ configure	29 Jul 2004 01:35:44 -0000
@@ -261,6 +261,7 @@
   --disable-sgiaudio     disable SGI sound support [autodetect]
   --disable-sunaudio     disable Sun sound support [autodetect]
   --disable-win32waveout disable Windows waveout sound support [autodetect]
+  --disable-openaudio    disable OpenAudio output for Darwin [autodetect]
   --disable-select       disable using select() on audio device [enable]
 
 Miscellaneous options:
@@ -1163,6 +1164,7 @@
 _sdl=auto
 _directx=auto
 _win32waveout=auto
+_openaudio=auto
 _nas=auto
 _png=auto
 _jpg=auto
@@ -1307,6 +1309,8 @@
   --disable-directx)    _directx=no     ;;
   --enable-win32waveout)  _win32waveout=yes ;;
   --disable-win32waveout) _win32waveout=no  ;;
+  --enable-openaudio)   _openaudio=yes  ;;
+  --disable-openaudio)  _openaudio=no   ;;
   --enable-nas)		_nas=yes	;;
   --disable-nas)	_nas=no		;;
   --enable-png)		_png=yes	;;
@@ -3903,6 +3907,27 @@
 fi
 echores "$_win32waveout"
 
+echocheck "OpenAudio"
+if test "$_openaudio" = auto ; then
+  cat > $TMPC << EOF
+#include <OpenAudio/audioio.h>
+#include <OpenAudio/defines.h>
+int main(void) { return 0; }
+EOF
+  _openaudio=no
+  cc_check -framework OpenAudio && _openaudio=yes
+fi
+if test "$_openaudio" = yes ; then
+  _def_openaudio='#define HAVE_OPENAUDIO 1'
+  _ld_openaudio="-framework OpenAudio"
+  _aosrc="$_aosrc ao_darwin.c"
+  _aomodules="darwin $_aomodules"
+else
+  _def_openaudio='#undef HAVE_OPENAUDIO'
+  _noaomodules="darwin $_noaomodules"
+fi
+echores "$_openaudio"
+
 echocheck "Directx"
 if test "$_directx" = auto ; then
   cat > $TMPC << EOF
@@ -6178,6 +6203,7 @@
 JACK_LIB = $_ld_jack
 JACK_INC = $_inc_jack
 SGIAUDIO_LIB = $_ld_sgiaudio
+OPENAUDIO_LIB = $_ld_openaudio
 
 # input/demuxer/codecs
 TERMCAP_LIB = $_ld_termcap
@@ -6588,6 +6614,7 @@
 $_def_sunaudio
 $_def_sgiaudio
 $_def_win32waveout
+$_def_openaudio
 $_def_nas
 
 /* Enable fast OSD/SUB renderer (looks ugly, but uses less CPU power) */
Index: libao2/audio_out.c
===================================================================
RCS file: /cvsroot/mplayer/main/libao2/audio_out.c,v
retrieving revision 1.39
diff -u -r1.39 audio_out.c
--- libao2/audio_out.c	10 Jul 2004 20:56:25 -0000	1.39
+++ libao2/audio_out.c	29 Jul 2004 01:35:44 -0000
@@ -53,6 +53,9 @@
 #ifdef HAVE_WIN32WAVEOUT
 extern ao_functions_t audio_out_win32;
 #endif
+#ifdef HAVE_OPENAUDIO
+extern ao_functions_t audio_out_darwin;
+#endif
 #ifdef HAVE_DXR2
 extern ao_functions_t audio_out_dxr2;
 #endif
@@ -90,6 +93,9 @@
 #ifdef USE_SUN_AUDIO
         &audio_out_sun,
 #endif
+#ifdef HAVE_OPENAUDIO
+	&audio_out_darwin,
+#endif
 // wrappers:
 #ifdef USE_ARTS
         &audio_out_arts,
